### PR TITLE
chore: upgrade OpenTelemetry Core to v0.55.0

### PR DESCRIPTION
**Pull request by sumo-drosiek**
upgrade OpenTelemetry Core to v0.55.0 as well as dependencies

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Please focus on markdown files

**Review by swiatekm-sumo — COMMENTED** [review]
Do we still need to use our `filterprocessor` fork?

**Comment by sumo-drosiek** [comment]
depends how we understand deprecation :sweat_smile: 
I can change it to breaking change and that way we can remove it immediately :)

File: docs/Upgrading.md [hunk 1]
```diff
@@ -13,6 +15,43 @@
...
+        match_type: expr
+        expressions:
+          - Body matches "log to include"
+      exclude:
+        match_type: expr
+        expressions:
+          - Body matches "log to exclude"
+```
+
+is equivalent of configuration of `logstransform` processor:
```
Comment (sumo-drosiek) [inline docs/Upgrading.md:40]:
```suggestion
is equivalent of the following configuration of `logstransform` processor:
```

**Comment by sumo-drosiek** [comment]
I added deprecation message to the filterprocessor:

```
2022/07/13 08:26:12 proto: duplicate proto type registered: jaeger.api_v2.PostSpansRequest
2022/07/13 08:26:12 proto: duplicate proto type registered: jaeger.api_v2.PostSpansResponse
2022-07-13T08:26:12.060+0200    info    service/telemetry.go:103        Setting up own telemetry...
2022-07-13T08:26:12.060+0200    info    service/telemetry.go:138        Serving Prometheus metrics      {"address": ":8888", "level": "basic"}
2022-07-13T08:26:12.061+0200    info    pipelines/pipelines.go:341      Component is under development. {"kind": "exporter", "data_type": "logs", "name": "logging", "stability": "in development"}
2022-07-13T08:26:12.061+0200    warn    filterprocessor@v0.55.0/filter_processor_logs.go:221
*********************************************************************************************************************************************************
***    Support for "expr" language is deprecated and is going to be dropped soon. Please see the migration document:                                  ***
***    https://github.com/SumoLogic/sumologic-otel-collector/blob/v0.55.0-sumo-0/docs/Upgrading.md#filter-processor-drop-support-for-expr-language.   ***
*********************************************************************************************************************************************************
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+This release deprecates the following change:
+
+- `filter` processor: drop support for `expr` language ([upgrade guide][upgrade_guide_v0_55_0_expr_support])
```
Comment (sumo-drosiek) [inline CHANGELOG.md:12]:
```suggestion
- `filter` processor: support for `expr` language ([upgrade guide][upgrade_guide_v0_55_0_expr_support])
```

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**